### PR TITLE
Renovate pinning and dependency upgrades

### PR DESCRIPTION
**Pull request by mzieniukbw**
## 📔 Objective

Dependencies upgrades and pinning:
- Remove `python` from dependency upgrades in Renovate
- Pin `python` to latest 3.7 - but allow for ranges for future Splunk Enterprise 9.3.0 compatibility, that requires python 3.9
- Upgraded `splunktaucclib`, `splunk-add-on-ucc-framework`, `splunk-appinspect` to newest
- Poetry lock file updated
- Updated GH Actions to build with python 3.9 - running on 3.7 on longer works (probably due to poetry not being compatible)

Regression tested on Splunk Enterprise 9.2.1 and Splunk Enterprise 9.3.0 beta - took the artifact from this PR

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by mzieniukbw** [comment]
@bitwarden/dept-devops @bitwarden/team-admin-console-dev I need an approval from a codeowners to merge this PR. Let me know if that's ok with you.

**Review by vincentsalucci — APPROVED** [review]
Thanks for taking this on @mzieniukbw

**Comment by mzieniukbw** [comment]
Thank you @withinfocus @vincentsalucci !

@bitwarden/dept-devops i need approval from you too to merge this PR :)

**Review by Eeebru — APPROVED** [review]

**Pull request merged**
